src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/documenttemplate/TemplateRetrievalServiceTest.kt

```diff
@@ -20,60 +25,310 @@ class TemplateRetrievalServiceTest {
 
   private val logAppender = findLogAppender(TemplateRetrievalService::class.java)
 
-  private val currentDateTime: ZonedDateTime = ZonedDateTime.now()
-  private val pastDocumentTemplateSetting = documentTemplateSettings(startDateTime = currentDateTime.minusMonths(2))
-  private val currentDocumentTemplateSetting =
-    documentTemplateSettings(startDateTime = currentDateTime.minusDays(2))
-  private val futureDocumentTemplateSetting =
-    documentTemplateSettings(startDateTime = currentDateTime.plusMonths(2))
+  private val currentDateTime: ZonedDateTime = ZonedDateTime.now(ZoneId.of("UTC"))
+  private val pastDateTime = currentDateTime.minusMonths(2)
+  private val presentDateTime = currentDateTime.minusDays(2)
+  private val futureDateTime = currentDateTime.plusMonths(2)
+  private val pastDocumentTemplateSetting = documentTemplateSettings(startDateTime = pastDateTime, "past")
+  private val currentDocumentTemplateSetting = documentTemplateSettings(startDateTime = presentDateTime, "present")
+  private val futureDocumentTemplateSetting = documentTemplateSettings(startDateTime = futureDateTime, "future")
+  private val defaultDocumentTemplateSetting = DocumentTemplateSetting(startDateTime = futureDateTime, "default")
 
-  private val documentTemplateSettingsList =
+  private val defaultSettingList =
     listOf(currentDocumentTemplateSetting, futureDocumentTemplateSetting, pastDocumentTemplateSetting)
 
-  @Test
-  fun `retrieves correct Part A template`() {
-    val documentTemplateConfiguration = documentTemplateConfiguration(
-      partATemplateSettings = documentTemplateSettingsList,
-    )
-    testLoadDocumentTemplate(documentTemplateConfiguration, DocumentType.PART_A_DOCUMENT)
+  private val documentTemplateConfiguration = documentTemplateConfiguration(
+    partATemplateSettings = defaultSettingList,
+    partAPreviewTemplateSettings = defaultSettingList,
+    dntrTemplateSettings = defaultSettingList,
+  )
+  private val noDocumentsTemplateConfiguration = documentTemplateConfiguration(
+    partATemplateSettings = listOf(),
+    partAPreviewTemplateSettings = listOf(),
+    dntrTemplateSettings = listOf(),
+  )
+  private val futureOnlyDocumentsTemplateConfiguration = documentTemplateConfiguration(
+    partATemplateSettings = listOf(futureDocumentTemplateSetting),
+    partAPreviewTemplateSettings = listOf(futureDocumentTemplateSetting),
+    dntrTemplateSettings = listOf(futureDocumentTemplateSetting),
+  )
+
+  val noExistingDocument: ZonedDateTime? = null
+
+  @Nested
+  @DisplayName("Part A Documents")
+  inner class PartADocument {
+    @Test
+    fun `retrieves the default Part A template when no settings provided`() {
+      testLoadDocumentTemplate(
+        noDocumentsTemplateConfiguration,
```

Comment:
Is it worth testing the cases with no config set but with both past and future existing dates, to check it still goes for the default?